modules/hostap/src/supp_api.c

```diff
@@ -2082,6 +2082,78 @@ int hapd_state(const struct device *dev, int *state)
 	return ret;
 }
 
+static int hapd_config_chan_center_seg0(struct wifi_connect_req_params *params)
+{
+	int ret = 0;
+	uint8_t center_freq_seg0_idx = 0;
+	uint8_t oper_chwidth = CHANWIDTH_USE_HT;
+	uint8_t *center_freq = NULL;
+	uint8_t center_freq_40MHz[] = {38, 46, 54, 62, 102, 110, 118, 126, 134, 142, 151, 159};
+	uint8_t center_freq_80MHz[] = {42, 58, 106, 122, 138, 155};
+	uint8_t index, index_max, chan_idx, ch_offset = 0;
+
+	/* Unless ACS is being used, both "channel" and "vht_oper_centr_freq_seg0_idx"
+	 * parameters must be set. */
+	switch (params->bandwidth) {
+	case WIFI_FREQ_BANDWIDTH_20MHZ:
+		oper_chwidth = CHANWIDTH_USE_HT;
+		center_freq_seg0_idx = params->channel;
+		break;
+	case WIFI_FREQ_BANDWIDTH_40MHZ:
+		if (!hostapd_cli_cmd_v("set ht_capab [HT40+][SHORT-GI-40]")) {
```

Comment:
SHORT-GI  should be set as per the underlying driver/chipset capablities

Reply:
the choice of HT40+/- should still be user configurable, no? 36-40MHz-HT40+ === 40-40MHz-HT40- == 38.

Reply:
When starting soft AP, it cannot get the driver/chipset capabilities. 
What do you think about putting these configurations in the "wifi ap config" command?
Before starting the AP, users can freely set it, similar to the hostapd.conf file.